tensorrt_llm/_torch/auto_deploy/shim/ad_executor.py

```diff
@@ -325,6 +325,12 @@ def create_autodeploy_executor(ad_config: LlmArgs, tokenizer: Optional[Tokenizer
     port = mpi_dist.broadcast(dist.get_free_port())  # use MPI broadcast to pick a free port
     dist.initialize_or_skip(rank, world_size, port)
 
+    # Log allreduce strategy if configured in transforms
+    if "detect_sharding" in ad_config.transforms:
+        strategy = ad_config.transforms["detect_sharding"].get("allreduce_strategy", "AUTO")
+        if strategy != "AUTO":
+            ad_logger.info(f"Using allreduce strategy from transforms config: {strategy}")
```

Comment:
how about we log this in the sharding transform? Seems more appropriate to do it there